ppafm/common.py

```diff
@@ -17,56 +19,61 @@
 AUMASS = 1.66053904020e-27  # [kg]
 
 
-# fmt: off
-# default parameters of simulation
-params = {
-    "PBC": True,
-    "nPBC": np.array([1, 1, 1]),
-    "gridN": np.array([-1, -1, -1]).astype(int),
-    "gridO": np.array([0.0, 0.0, 0.0]),
-    "gridA": np.array([0.0, 0.0, 0.0]),
-    "gridB": np.array([0.0, 0.0, 0.0]),
-    "gridC": np.array([0.0, 0.0, 0.0]),
-    "FFgrid0": np.array([-1.0, -1.0, -1.0]),
-    "FFgridA": np.array([-1.0, -1.0, -1.0]),
-    "FFgridB": np.array([-1.0, -1.0, -1.0]),
-    "FFgridC": np.array([-1.0, -1.0, -1.0]),
-    "moleculeShift": np.array([0.0, 0.0, 0.0]),
-    "probeType": "O",
-    "charge": 0.00,
-    "Apauli": 18.0,
-    "Bpauli": 1.0,
-    "ffModel": "LJ",
-    "Rcore": 0.7,
-    "r0Probe": np.array([0.00, 0.00, 4.00]),
-    "stiffness": np.array([-1.0, -1.0, -1.0]),
-    "klat": 0.5,
-    "krad": 20.00,
-    "tip": "s",
-    "sigma": 0.7,
-    "scanStep": np.array([0.10, 0.10, 0.10]),
-    "scanMin": np.array([0.0, 0.0, 5.0]),
-    "scanMax": np.array([20.0, 20.0, 8.0]),
-    "scanTilt": np.array([0.0, 0.0, -0.1]),
-    "tiltedScan": False,
-    "kCantilever": 1800.0,
-    "f0Cantilever": 30300.0,
-    "Amplitude": 1.0,
-    "plotSliceFrom": 16,
-    "plotSliceTo": 22,
-    "plotSliceBy": 1,
-    "imageInterpolation": "bicubic",
-    "colorscale": "gray",
-    "colorscale_kpfm": "seismic",
-    "ddisp": 0.05,
-    "aMorse": -1.6,
-    "tip_base": np.array(["None", 0.00]),
-    "Rtip": 30.0,
-    "permit": 0.00552634959,
-    "vdWDampKind": 2,
-    "#": None,
-}
-# fmt: on
+class PpafmParameters(pydantic.BaseModel):
+    """Parameters for the PPAFM simulation.
+
+    The parameters description can be found on the wiki page: https://github.com/Probe-Particle/ppafm/wiki/Setting-simulation-parameters
+    """
+
+    PBC: bool = True
+    nPBC: typing.List[int] = [1, 1, 1]
+    gridN: typing.List[int] = [-1, -1, -1]
+    gridO: typing.List[float] = [0.0, 0.0, 0.0]
+    gridA: typing.List[float] = [20.0, 0.0, 0.0]
+    gridB: typing.List[float] = [0.0, 20.0, 0.0]
+    gridC: typing.List[float] = [0.0, 0.0, 20.0]
+    FFgrid0: typing.List[float] = [-1.0, -1.0, -1.0]
+    FFgridA: typing.List[float] = [-1.0, -1.0, -1.0]
+    FFgridB: typing.List[float] = [-1.0, -1.0, -1.0]
+    FFgridC: typing.List[float] = [-1.0, -1.0, -1.0]
+    moleculeShift: typing.List[float] = [0.0, 0.0, 0.0]
+    probeType: str = "O"
+    charge: float = 0.00
+    Apauli: float = 18.0
+    Bpauli: float = 1.0
+    ffModel: str = "LJ"
+    Rcore: float = 0.7
+    r0Probe: typing.List[float] = [0.00, 0.00, 4.00]
+    stiffness: typing.List[float] = [-1.0, -1.0, -1.0]
+    klat: float = 0.5
+    krad: float = 20.00
+    tip: str = "s"
```

Comment:
`tip` can also sometimes be an `np.ndarray`. This is for example the case in `generateElFF.py` when the tip density is passed as a `.xsf` file. Can this be a `typing.Union[str, np.ndarray]`?

Reply:
done in https://github.com/Probe-Particle/ppafm/pull/275/commits/fe8eb2200d13462bcb1a401dc7087e1a1e83fd7b and in https://github.com/Probe-Particle/ppafm/pull/275/commits/42397888c04666910d8e2571e08c1c9204f4ffb2